src/Storages/AlterCommands.cpp

```diff
@@ -529,9 +530,51 @@ void AlterCommand::apply(StorageInMemoryMetadata & metadata, ContextPtr context)
         {
             metadata.columns.add(column, after_column, first);
         }
+
+        /// Try to add "implicit" minmax index for new column
+        if ((isNumber(column.type) && metadata.settings_changes && metadata.settings_changes->as<ASTSetQuery &>().changes.tryGet("add_minmax_index_for_numeric_columns"))
+         || (isString(column.type) && metadata.settings_changes && metadata.settings_changes->as<ASTSetQuery &>().changes.tryGet("add_minmax_index_for_string_columns")))
+        {
+            bool minmax_index_exists = false;
+            for (const auto & index: metadata.secondary_indices)
+            {
+                if (index.column_names.front() == column.name && index.type == "minmax")
+                {
+                    minmax_index_exists = true;
+                    break;
+                }
+            }
+
+            if (!minmax_index_exists)
+            {
+                auto index_type = makeASTFunction("minmax");
+                auto index_ast = std::make_shared<ASTIndexDeclaration>(std::make_shared<ASTIdentifier>(column.name), index_type, IMPLICITLY_ADDED_MINMAX_INDEX_PREFIX + column.name);
+                metadata.secondary_indices.push_back(IndexDescription::getIndexFromAST(index_ast, metadata.columns, context));
+            }
+        }
     }
     else if (type == DROP_COLUMN)
     {
+        const auto * column = metadata.columns.tryGet(column_name);
```

Comment:
It is not clear to me why l. 558-576 are necessary? We are dropping a column, does l. 580 not drop all existing indexes on the dropped column already? It would be interesting to check what happens with normal secondary indexes if DROP column is ran (there is no special logic for this case).

Reply:
ya you are right, its not needed. Removed it, thanks

Reply:
Sorry looks like it doesn't drop index , added the logic back. We get '`Cannot apply mutation because it breaks skip index`'